### DISCUSSION TITLE
Buggy experience / pseudo classes not working / styles often won't be applied / intellisense

### DISCUSSION THREAD

**Discussion by exotexot** [nativewind/nativewind#250] [General]
Hey,

just started using native wind.
I do have a poor developer experience though. Often times styles won't be applied and I need to restart / rebuild the entire app. 

Also I have pseudo classes like hover not working. Are there any limitations ? e.g. https://github.com/marklawlor/nativewind/issues/170 this code is not working for me.

Also the intelligence in my VSCode is buggy. Often there are no suggestions made.

I've read the documents and setup guide.

Using react-native 0.7.0 with typescript. 
nativewind 2.0.10

**Comment by exotexot** [comment]
Have you tested nativewind on macOS via Catalyst? 
also responsive pseudo classes only apply after the App window blurs and focuses again...

macOS: pseudo class isn't working. I guess this requires a workaround of some sorts. Maybe use react-native-device-info as it detects Catalyst.

**Comment by marklawlor** [comment]
Moving this to a discussion as this doesn't seem isolated around a single issue. I'm happy to create new issues from this if needed.

> Often times styles won't be applied and I need to restart / rebuild the entire app

Yep, there's a pinned issue right now on this topic - I'm rebuilding a lot of things to make this work a lot better. Should have updates soon.

> Also I have pseudo classes like hover not working. Are there any limitations ?

Hover only works on React Native platforms that support the onMouse event listeners. I believe this is only Web, Windows and OSX. 

You follow up by saying you are using Catalyst. I believe Catalyst is simply an iOS emulator and as iOS does not have hover events, your desktop app will not have them either.

> Also the intelligence in my VSCode is buggy. Often there are no suggestions made.

NativeWind doesn't touch/control/do anything to the extension. If you have any problems with it, it most likely cannot parse your file due to a syntax error. I'd suggest checking the extension logs and consulting the extension issues for more information.

> Have you tested nativewind on macOS via Catalyst?

No I have not and I would consider it outside the current scope of the project. Same with any other mobile emulator.

> also responsive pseudo classes only apply after the App window blurs and focuses again...

NativeWind's responsive classes work by using [`Dimention.addEventListener`](https://reactnative.dev/docs/dimensions#addeventlistener). You should test if RN's Dimension class still fires the `change` event while unfocused. If it does not, then NativeWind will not be able to apply the responsive styles until focused again.

> macOS: pseudo class isn't working. 

This uses [`Platform.OS`](https://reactnative.dev/docs/platform#os) for detection. I assume the Catalyst is reporting the OS as `iOS`.

The `macos` variant was designed for https://github.com/microsoft/react-native-macos

Hope that clears some things up for you.

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
